cpp/autosar/src/rules/M16-1-1/DefinedPreProcessorOperatorGeneratedFromExpansionFound.ql

```diff
@@ -16,16 +16,19 @@ import cpp
 import codingstandards.cpp.autosar
 import DefinedMacro
 
-from DefinedMacro m, PreprocessorBranch e
+/**
+ * An `if` or `elif` preprocessor branch.
+ */
+class PreprocessorIfOrElif extends PreprocessorBranch {
```

Comment:
we have this concept replicated quite a few times it seems ([here](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/rules/undefinedmacroidentifiers/UndefinedMacroIdentifiers.qll#L79), [here](https://github.com/github/codeql-coding-standards/blob/main/cpp/autosar/src/rules/M16-1-1/DefinedPreProcessorOperatorInOneOfTheTwoStandardForms.ql#L40)), should we consider placing the class in a more [central location](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/PreprocessorDirective.qll)?

Reply:
Great spot! I also found one more instance (in `RULE-20-8`) and have addressed all three cases.